From 4ed83e06e330bf3a225096f7db891293c7023834 Mon Sep 17 00:00:00 2001
From: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Date: Fri, 11 Aug 2023 12:11:39 +0200
Subject: [PATCH 0510/1141] arm: dts: st: fix DSI peripheral clock on stm32mp15
 boards

In RCC driver, 'DSI_K' is a kernel clock while 'DSI' has pclk4 as parent
clock, which means that it is an APB peripheral clock. Swap the clocks
in the DSI peripheral clock reference.

Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Change-Id: I71e98719b1ad033fa2f36b5026791b8b1eac0336
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/322702
Reviewed-by: Yannick FERTRE <yannick.fertre@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yannick FERTRE <yannick.fertre@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm/boot/dts/stm32mp157.dtsi           | 2 +-
 arch/arm/boot/dts/stm32mp157a-dk1-scmi.dtsi | 2 +-
 arch/arm/boot/dts/stm32mp157a-ed1-scmi.dtsi | 2 +-
 arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi | 2 +-
 arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi | 2 +-
 arch/arm/boot/dts/stm32mp157c-ed1-scmi.dtsi | 2 +-
 arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi | 2 +-
 7 files changed, 7 insertions(+), 7 deletions(-)

--- a/arch/arm/boot/dts/stm32mp157.dtsi
+++ b/arch/arm/boot/dts/stm32mp157.dtsi
@@ -21,7 +21,7 @@
 			compatible = "st,stm32-dsi";
 			reg = <0x5a000000 0x800>;
 			phy-dsi-supply = <&reg18>;
-			clocks = <&rcc DSI_K>, <&clk_hse>, <&rcc DSI_PX>;
+			clocks = <&rcc DSI>, <&clk_hse>, <&rcc DSI_PX>;
 			clock-names = "pclk", "ref", "px_clk";
 			resets = <&rcc DSI_R>;
 			reset-names = "apb";
--- a/arch/arm/boot/dts/stm32mp157a-dk1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157a-dk1-scmi.dtsi
@@ -24,7 +24,7 @@
 };
 
 &dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+	clocks = <&rcc DSI>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
 &gpioz {
--- a/arch/arm/boot/dts/stm32mp157a-ed1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157a-ed1-scmi.dtsi
@@ -24,7 +24,7 @@
 };
 
 &dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+	clocks = <&rcc DSI>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
 &gpioz {
--- a/arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157a-ev1-scmi.dtsi
@@ -22,7 +22,7 @@
 };
 
 &dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+	clocks = <&rcc DSI>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
 &gpioz {
--- a/arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-dk2-scmi.dtsi
@@ -29,7 +29,7 @@
 };
 
 &dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+	clocks = <&rcc DSI>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
 &gpioz {
--- a/arch/arm/boot/dts/stm32mp157c-ed1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-ed1-scmi.dtsi
@@ -29,7 +29,7 @@
 };
 
 &dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+	clocks = <&rcc DSI>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
 &gpioz {
--- a/arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-ev1-scmi.dtsi
@@ -27,7 +27,7 @@
 };
 
 &dsi {
-	clocks = <&rcc DSI_K>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
+	clocks = <&rcc DSI>, <&scmi_clk CK_SCMI_HSE>, <&rcc DSI_PX>;
 };
 
 &gpioz {
